### PR TITLE
[CoreBundle] Registers last view mode of resources manager

### LINKED ISSUE #207: Gestionnaire de ressource : vue liste
Mémoriser la demande d'une vue (liste ou icone) et afficher cette vue par défaut lors du retour dans l'espace de ressource

### PR DESCRIPTION
**Pull request by kitan1982**
- Registers last view mode of resources manager
- closes #207

### CODE REVIEW

File: main/core/Controller/ResourceController.php [hunk 1]
```diff
@@ -1135,6 +1140,25 @@ public function importAction(ResourceNode $directory)
...
+     *     name="claro_resource_manager_display_mode_register",
+     *     options={"expose"=true}
+     * )
+     * @EXT\ParamConverter("user", converter="current_user", options={"allowAnonymous"=true})
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function resourceManagerDisplayModeRegisterAction(User $user, $index, $displayMode)
+    {
+        if ($user !== 'anon.') {
```
Comment (stefk) [inline main/core/Controller/ResourceController.php]: That condition will always be true due to the type hinting above.

Comment (stefk) [inline main/core/Controller/ResourceController.php]: Correct type hinting should probably be: `User $user = null`

Comment (stefk) [inline main/core/Controller/ResourceController.php]: (and the current_user converter doesn't return '.anon')

File: main/core/Controller/ResourceController.php [hunk 2]
```diff
@@ -1135,6 +1140,25 @@ public function importAction(ResourceNode $directory)
...
+            $this->userManager->registerResourceManagerDisplayModeByUser($user, $index, $displayMode);
+        }
+
+        return new Response(200);
+    }
+
     public function deleteNodeConfirmAction(ResourceNode $node)
     {
         throw new \Exception('hey');
```
Comment (stefk) [inline main/core/Controller/ResourceController.php]: Not your code but seems like garbage.

File: main/core/Resources/public/js/resource/manager/views/resource/actions.js [hunk 3]
```diff
@@ -335,7 +337,7 @@
                 && this.isReadyToPaste
                 && (!this.isCutMode || this.checkedNodes.directoryId !== event.id);
 
-            var listViewActivated = this.displayMode === 'default' ? false: true;
+            var listViewActivated = this.displayMode === 'list' ? true: false;
```
Comment (stefk) [inline main/core/Resources/public/js/resource/manager/views/resource/actions.js]: Should be simplified to:

```php
var listViewActivated = this.displayMode === 'list';
```

**Comment by LaurentGruber** [comment]
LGTM

**Comment by kitan1982** [comment]
LGTM

**Pull request merged**